routers/products.js

```diff
@@ -11,9 +22,49 @@ router.get("/:productId", (req, res) => {
   const productId = Number(req.params.productId);
   const product = products.find(({ id }) => id === productId);
   if (!product) {
-    return res.status(404).send("Product not found");
+    throw new NotFoundError();
+  }
+  res.status(200).json(product);
+});
+
+router.post("/", (req, res) => {
+  const { name, description, category, price } = req.body;
+  const product = { id: randomUUID(), name, description, category, price };
+  try {
+    const data = fs.readFileSync(productsStoreFilePath, "utf-8");
+    const parsedData = JSON.parse(data === "" ? [] : data);
+    parsedData.push(product);
+    fs.writeFileSync(productsStoreFilePath, `${JSON.stringify(parsedData)}`);
+  } catch (err) {
+    throw new Error(err.message);
   }
   res.status(200).json(product);
 });
 
+router.post("/import", (req, res) => {
+  const writableStream = fs.createWriteStream(productsStoreFilePath);
+  let lastItem = null;
+
+  eventEmitter.emit("fileUploadStart");
+
+  req
+    .pipe(csv())
+    .on("data", (data) => {
+      writableStream.write(
+        lastItem == null ? "[" : `${JSON.stringify(lastItem)},`
+      );
+      lastItem = data;
+    })
+    .on("end", () => {

```

Comment:
This `end` event is attached to the Transform stream(csv()), not to Writable that is the latest

Reply:
changed